Automattic/wp-calypso
# Reader: Support Jetpack story block in shared css and open in a new tab on click

Alternative implementation to #44985 to support the mobile readers as well as Calypso. It relies on the shared reader css logic introduced in https://github.com/Automattic/wp-calypso/pull/43073.

#### Changes proposed in this Pull Request

@Tug we discussed this a bit and we're not big fans of duplicating the css from the block, but this seems to me the best way to support all the readers well.

I looked at doing things like `@import url("https://s0.wp.com/wp-content/mu-plugins/jetpack/_inc/blocks/story/view.css")` in the scss file itself, but this has two problems:

1. It includes quite a lot of unnecessary css only used for the fullscreen player (and not relevant to the reader).
2. It looks like the css isn't inlined when included from a URL, so instead another css file download is triggered. This isn't so bad for Calypso but the mobile readers rely on caching the reader-mobile.css, and the caching rules won't apply to that file.

2 surprised me a bit, when inlining a local file the css is rendered and included in the outputed reader-mobile.css, but apparently not for remote URLs. If there's a way to do that let me know!

As well as moving the CSS, I also moved the fullscreen/autoplay query parameters to the block renderer.

I did keep the `target=_blank` logic in Calypso because the mobile readers don't need it and it feels to me more like a reader concern. Happy to move that though if you think that makes more sense.

Note: Some small changes are needed in the mobile readers themselves. I'll open PRs for those too, but I'd like to know if this is headed in the right direction first.

<img width="1036" alt="Screen Shot 2020-09-04 at 10 47 53 AM" src="https://user-images.githubusercontent.com/9613966/92190435-2a49a600-ee9c-11ea-88c9-37b0050455b6.png">

#### Testing instructions

* Open the reader at http://calypso.localhost:3000/
* Subscribe to a site that has stories and has https://github.com/Automattic/jetpack/pull/17069 applied.
* Open a story in full page, confirm the CSS is applied and the preview looks like the screenshot.
* Click on the embed, a new tab should open and the story should load in fullscreen.
